ddtrace/_trace/processor/__init__.py

```diff
@@ -324,6 +324,14 @@ def on_span_finish(self, span):
 
                 if should_partial_flush:
                     log.debug("Partially flushing %d spans for trace %d", num_finished, span.trace_id)
+                    if not finished:
+                        log_msg = "Partial flush triggered but no spans to flush (was tracer reconfigured?)"
```

Comment:
are we able to include the span repr in the message? that might help us narrow down when/where the bug is being triggered.

e.g. for your testing in CI it was the pytest test suite span.

Reply:
Test `stderr` says:
```
No spans to flush in partial flush for span: <Span(id=15838273028537712397,trace_id=135991983792686704567357058663853114310,parent_id=None,name=regression)> (was tracer reconfigured?)
```